tools/ldb_cmd.cc

```diff
@@ -2842,10 +2915,61 @@ class InMemoryHandler : public WriteBatch::Handler {
   const std::map<uint32_t, const Comparator*> ucmps_;
 };
 
-void DumpWalFile(Options options, std::string wal_file, bool print_header,
-                 bool print_values, bool is_write_committed,
+void DumpWalFiles(Options options, const std::string& dir_or_file,
+                  bool print_header, bool print_values,
+                  bool only_print_seqno_gaps, bool is_write_committed,
+                  const std::map<uint32_t, const Comparator*>& ucmps,
+                  LDBCommandExecuteResult* exec_state) {
+  std::vector<std::string> filenames;
+  ROCKSDB_NAMESPACE::Env* env = options.env;
+  ROCKSDB_NAMESPACE::Status st = env->GetChildren(dir_or_file, &filenames);
+  std::optional<SequenceNumber> prev_batch_seqno = std::nullopt;
+  std::optional<uint32_t> prev_batch_count = std::nullopt;
+  if (!st.ok() || filenames.empty()) {
+    // dir_or_file does not exist or does not contain children
+    // Check its existence first
+    Status s = env->FileExists(dir_or_file);
+    // dir_or_file does not exist
+    if (!s.ok()) {
+      if (exec_state) {
+        *exec_state = LDBCommandExecuteResult::Failed(
+            dir_or_file + ": No such file or directory");
+      }
+      return;
+    }
+    DumpWalFile(options, dir_or_file, print_header, print_values,
+                only_print_seqno_gaps, is_write_committed, ucmps, exec_state,
+                &prev_batch_seqno, &prev_batch_count);
```

Comment:
Question: what happens here if `dir_or_file` is an existing but empty directory?

Reply:
Good catch! The error message is not clear, I have added checks for it to have a legit log file name.